src/dawn/IIR/MetaInformation.h

```diff
@@ -0,0 +1,493 @@
+  /// Map of AccessIDs and to the name of the variable/field. Note that only for fields of the "main
+  /// stencil" we can get the AccessID by name. This is due the fact that fields of different
+  /// stencil functions can share the same name.
+  std::unordered_map<std::string, int> NameToAccessIDMap_;
+  std::unordered_map<int, std::string> AccessIDToNameMap_;
+
+  /// Surjection of AST Nodes, Expr (FieldAccessExpr or VarAccessExpr) or Stmt (VarDeclStmt), to
+  /// their AccessID. The surjection implies that multiple AST Nodes can have the same AccessID,
+  /// which is the intended behaviour as we want to get the same ID back when we access the same
+  /// field for example
+  std::unordered_map<std::shared_ptr<Expr>, int> ExprToAccessIDMap_;
+  std::unordered_map<std::shared_ptr<Stmt>, int> StmtToAccessIDMap_;
+
+  /// Injection of AccessIDs of literal constant to their respective name (usually the name is just
+  /// the string representation of the value). Note that literals always have *strictly* negative
+  /// AccessIDs, which makes them distinguishable from field or variable AccessIDs. Further keep in
+  /// mind that each access to a literal creates a new AccessID!
+  std::unordered_map<int, std::string> LiteralAccessIDToNameMap_;
+
+  /// This is a set of AccessIDs which correspond to fields. This allows to fully identify if a
+  /// AccessID is a field, variable or literal as literals have always strictly negative IDs and
+  /// variables are neither field nor literals.
+  std::set<int> FieldAccessIDSet_;
+
+  /// This is an ordered list of IDs of fields that belong to the user API call of the program
+  std::vector<int> apiFieldIDs_;
+
+  /// Set containing the AccessIDs of fields which are represented by a temporary storages
+  std::set<int> TemporaryFieldAccessIDSet_;
+
+  /// Set containing the AccessIDs of fields which are manually allocated by the stencil and serve
+  /// as temporaries spanning over multiple stencils
+  std::set<int> AllocatedFieldAccessIDSet_;
+
+  /// Set containing the AccessIDs of "global variable" accesses. Global variable accesses are
+  /// represented by global_accessor or if we know the value at compile time we do a constant
+  /// folding of the variable
+  std::set<int> GlobalVariableAccessIDSet_;
+
+  /// Map of AccessIDs to the list of all AccessIDs of the multi-versioned field, for fields and
+  /// variables
+  VariableVersions variableVersions_;
+
+  /// Stencil description statements. These are built from the StencilDescAst of the sir::Stencil
+  std::vector<std::shared_ptr<Statement>> stencilDescStatements_;
+  std::unordered_map<std::shared_ptr<StencilCallDeclStmt>, int> StencilCallToStencilIDMap_;
+  std::unordered_map<int, std::shared_ptr<StencilCallDeclStmt>> IDToStencilCallMap_;
+
+  /// StageID to name Map. Filled by the `PassSetStageName`.
+  std::unordered_map<int, std::string> StageIDToNameMap_;
```

Comment:
it looks to me like this does not belong here. It is redundant, the information of the name of the stage is already in the Stage

Reply:
fixed